app/assets/stylesheets/common/base/d-image-grid.scss

```diff
@@ -99,3 +99,162 @@
     }
   }
 }
+
+.d-image-carousel {
+  position: relative;
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  gap: var(--space-1);
+
+  &__track {
+    display: flex;
+    width: 100%;
+    height: 400px;
+    overflow-x: auto;
+    overflow-y: hidden;
+    scroll-snap-type: x mandatory;
+    scroll-behavior: smooth;
+    -webkit-overflow-scrolling: touch;
+    scrollbar-width: none;
+
+    &::-webkit-scrollbar {
+      display: none;
+    }
+  }
+
+  &__slide {
+    .--carousel & {
+      flex: 0 0 100%;
+      width: 100%;
+      height: 100%;
+      scroll-snap-align: center;
+      scroll-snap-stop: always;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+    }
+
+    img:not(.thumbnail, .ytp-thumbnail-image, .emoji) {
+      max-width: 100% !important;
+      max-height: 100% !important;
+      width: auto !important;
+      height: auto !important;
+      object-fit: contain !important;
+      border-radius: var(--d-border-radius);
+      transition: box-shadow 0.4s ease;
```

Comment:
```suggestion
```

Tiny nitpick, but there is no `box-shadow` on the element, so, no need to transition.